### PR TITLE
Implementation of generate_secret

### PR DESCRIPTION
**Pull request by kevinburchfield**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,14 +1,98 @@
...
+) -> String {
+    let byte_array_length = match length {
+        Some(l) => l,
+        None => 32 
+    };
+
+    let byte_array: Vec<u8> = (0..byte_array_length).map(|_| { rand::random::<u8>() }).collect();
+
+    // Static for efficiency rather than leaving it to runtime to init a vector of chars
+    static CHAR_SET: [char; 62] = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
```
Comment (JayHelton) [inline src/lib.rs]: should this be lifted up out of the function?

File: src/lib.rs [hunk 2]
```diff
@@ -1,14 +1,98 @@
...
-pub fn generate_secret(){}
+
+pub fn generate_secret(
+    length: Option<u32>,
+    symbols: Option<bool>,
+    otpauth_url: Option<bool>,
+    name: Option<&str>,
+    issuer: Option<&str>,
+) -> SecretKey {
+    SecretKey { ascii: None, hex: None, base32: None, otpauth_url: None } // TODO (kevinburchfield) - fix. Only did this so it would compile
```
Comment (JayHelton) [inline src/lib.rs]: good for now, but after looking at the function, i think having this generate an otpauth url is a little odd. Id rather split that functionality up into separate functions. What do you think?

**Review by JayHelton — APPROVED** [review]

File: src/lib.rs (hunk 2 above)
Comment (kevinburchfield) [reply src/lib.rs]: Yeah, that makes sense. speakeasy has the generateSecret function just calling another function if that option is passed so it's no problem to pull it out.

Comment (kevinburchfield) [reply src/lib.rs]: Pulled it out and stubbed the otpauth function

**Comment by JayHelton** [comment]
ship it!

**Pull request merged**